xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -153,6 +155,30 @@ def verify_load_bounds(cast: CastOp, load: LoadOp):
         )
 
 
+class IndexOpToLoopSSA(RewritePattern):
+    @staticmethod
+    def discover_enclosing_loops(op: Operation) -> Sequence[Operation]:
+        loop_list: Sequence[Operation] = []
+        parent_op = op.parent_op()
+        if parent_op is not None:
+            loop_list.extend(IndexOpToLoopSSA.discover_enclosing_loops(parent_op))
+        if isa(op, scf.For) or isa(op, scf.ParallelOp):
+            loop_list.append(op)
+        return loop_list
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: IndexOp, rewriter: PatternRewriter, /):
+        # TODO: This does not currently support the offset in indexop, we can
```

Comment:
Maybe add an assert instead of a TODO? (If this is straightforward to check)

Reply:
Good idea, thanks - I've updated it to assert that all offsets are zero (as otherwise it won't be handled)